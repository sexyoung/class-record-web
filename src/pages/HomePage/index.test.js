import { render, cleanup, getByText } from '@testing-library/react';
import UserEvent from "@testing-library/user-event";
import { BrowserRouter } from "react-router-dom";
import { HomePage } from '.';

afterEach(cleanup);


it('should first', () => {
  const { container } = render(<BrowserRouter><HomePage /></BrowserRouter>);
  expect(container).toHaveTextContent('上課點點名');
  expect(container).toHaveTextContent('老師註冊');
  expect(container).toHaveTextContent('已註冊');
  expect(container).toHaveTextContent('點此登入');
  expect(container).toHaveTextContent('掌握學生出席狀況');
  expect(container).toHaveTextContent('終生免費');
  expect(container).toHaveTextContent('職人推薦');
  expect(container).toMatchSnapshot();
});

// it('should go to sign-up page', () => {
//   const { container, getByText } = render(<BrowserRouter><HomePage /></BrowserRouter>);
//   UserEvent.click(getByText('老師註冊'));
//   expect(container).toHaveTextContent('已有帳號');
// });
