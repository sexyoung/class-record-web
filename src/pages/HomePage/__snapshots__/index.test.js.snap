// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should first 1`] = `
<div>
  <div
    class="HomePage"
  >
    <div
      class="hero"
    >
      <svg
        aria-hidden="true"
        class="logo logo"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 5H7a2 2 0 00-2 2v12a2 2 0 002 2h10a2 2 0 002-2V7a2 2 0 00-2-2h-2M9 5a2 2 0 002 2h2a2 2 0 002-2M9 5a2 2 0 012-2h2a2 2 0 012 2m-6 9l2 2 4-4"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
      <p
        class="title"
      >
        上課點點名
      </p>
      <p
        class="subtitle"
      >
        輕鬆點名 優雅管理
      </p>
      <a
        class="btn signup"
        href="/sign-up"
      >
        老師註冊
      </a>
      <p
        class="signed"
      >
        已註冊？
        <a
          class="login"
          href="/login"
        >
          點此登入
        </a>
      </p>
    </div>
    <div
      class="block"
    >
      <h2
        class="title"
      >
        掌握學生出席狀況
      </h2>
      <div
        class="hint"
      >
        這邊放一張圖
      </div>
      <p
        class="text"
      >
        郭台銘說，關於BNT疫苗到貨將比原定時程更快更多，進度超前，相信中央流行疫情指揮中心也將很快會開放，有意願施打BNT的民眾都能排到，也會在原廠建議施打第二劑的時間點，安排後續接種進度。
      </p>
    </div>
    <div
      class="block"
    >
      <h2
        class="title"
      >
        終生免費
      </h2>
      <p
        class="text"
      >
        是的沒錯，我們的服務完全免費。基本的功能即可完成你手邊的點名與管理工作。如果想要更進階的服務，我們也提供以下進階功能！
      </p>
    </div>
    <div
      class="block"
    >
      <h2
        class="title"
      >
        職人推薦
      </h2>
      <div
        class="referrer"
      >
        <div
          class="img"
          style="background-image: url(grace.png);"
        />
        <div
          class="name"
        >
          Grace
        </div>
        <div
          class="title"
        >
          BetweenGos 職場才女 共同創辦人
        </div>
        <div
          class="text"
        >
          鴻鈞老祖望了一眼眼神冷冽的林坤，雙目之中，不由的掠過一絲震撼。 他怎麼也沒有想到，自己作爲天道化身，竟然都沒有保護準提周全。 不過，看準提這幅模樣，第十八重天四周觀戰的個仙府修士們，和嫦娥、王母等人，則都心中很是舒暢。 畢竟，西方教仗着西方二聖，在三界爲非作歹多年，早已引的一衆修士敢怒不敢言，而今林坤也算是替他們主持公道了。
        </div>
      </div>
      <div
        class="referrer"
      >
        <div
          class="img"
          style="background-image: url(kelly.png);"
        />
        <div
          class="name"
        >
          詩詩
        </div>
        <div
          class="title"
        >
          日本流行文化研究
        </div>
        <div
          class="text"
        >
          準提道人目光之中，閃過一抹狠辣，此刻的他，居然當着林坤的面，在鴻鈞老祖面前告起狀來。 不過，在他聲音響起的同時，鴻鈞老祖的臉色，卻是陡然一變，望向他的目光之中，閃過一抹厭惡。 顯然，雖然他召喚準提，前來加入對林坤的魔鬼試煉，卻並沒有真正拿他當做自己人。 就連方纔自己出手救他，也是因爲天命使然，準提不能這麼早就隕落，之後的三界大勢中，還有他的作用而已。
        </div>
      </div>
      <div
        class="referrer"
      >
        <div
          class="img"
          style="background-image: url(sexyoung.png);"
        />
        <div
          class="name"
        >
          Sexyoung
        </div>
        <div
          class="title"
        >
          音樂遊戲攻略指導
        </div>
        <div
          class="text"
        >
          看來經過這次的浩劫，他再想和林坤一對一的戰鬥，已經沒有這個資格了！ 從他望向林坤的目光中，就可以看出，林坤在他的心中，留下了多大的陰影。 “天地共主魔鬼試煉已結束，從今日起，林坤，便是這九霄之內，新的天地共主，和我鴻鈞的第四位徒弟！” 鴻鈞老祖一手捻鬚，盤坐蒲團之上，望着已然褪去鯤鵬之身，傲然立於虛空之中的林坤，朗聲說道。 “爲了表彰他，今日，我便破例在此開壇講法，希望大家都認真聽！” 在他聲音響起的同時，虛空之中，一道道璀璨的金色蓮花，在一個個修士腳下，次第開放。
        </div>
      </div>
    </div>
  </div>
</div>
`;
